ui-admin/src/portal/siteContent/HtmlSectionEditor.tsx

```diff
@@ -27,26 +28,30 @@ const HtmlSectionEditor = ({
   updatePage,
   section,
   sectionIndex,
+  siteInvalid,
+  setSiteInvalid,
   readOnly
 }: {
   htmlPage: HtmlPage,
-  updatePage: (page: HtmlPage) => void,
+  updatePage: (page: HtmlPage) => void
   section: HtmlSection
   sectionIndex: number
+  siteInvalid: boolean
+  setSiteInvalid: (invalid: boolean) => void
   readOnly: boolean
 }) => {
-  const sectionConfig = JSON.stringify(JSON.parse(section?.sectionConfig ?? '{}'), null, 2)
+  const [sectionContainsErrors, setSectionContainsErrors] = useState(false)
   const initial = SECTION_TYPES.find(sectionType => sectionType.value === section.sectionType)
   const [sectionTypeOpt, setSectionTypeOpt] = useState(initial)
 
-  const updateSection = (sectionIndex: number, updatedSection: HtmlSection) => {
-    try {
-      JSON.parse(updatedSection.sectionConfig ?? '{}')
-    } catch (e) {
-      // for now, we just don't allow changing the object structure itself -- just plain text edits
-      return
-    }
+  const [editorValue, setEditorValue] = useState(() =>
+    JSON.stringify(JSON.parse(section?.sectionConfig ?? '{}'), null, 2))
 
+  useEffect(() => {
+    setEditorValue(JSON.stringify(JSON.parse(section?.sectionConfig ?? '{}'), null, 2))
```

Comment:
I inadvertently started testing an earlier version of this PR where this section had a _setEditorValue and then a useCallback.  I'm glad you took the extra time to clean this up!